esmvaltool/diag_scripts/weighting/weighted_temperature_map.py

```diff
@@ -0,0 +1,188 @@
+"""Implementation of a mapplot for the climwip weighting scheme.
+
+Lukas Brunner et al. section 2.4
+https://iopscience.iop.org/article/10.1088/1748-9326/ab492f
+"""
+import logging
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
+import numpy as np
+import xarray as xr
+from climwip import log_provenance, read_model_data
+from climwip import get_diagnostic_filename, get_plot_filename
+
+from esmvaltool.diag_scripts.shared import run_diagnostic
+from esmvaltool.diag_scripts.weighting.plot_utilities import (
+    read_weights,
+    read_metadata,
+    weighted_quantile,
+)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def mapplot(dataarray, cfg, title_pattern, filename_part, ancestors, cmap=None, center=None):
+    """Visualize weighted temperature."""
+    period = '{start_year}-{end_year}'.format(**read_metadata(cfg)['tas'][0])
+    if 'tas_reference' in read_metadata(cfg).keys():
+        meta = read_metadata(cfg)['tas_reference']
+        period = 'change: {} minus {start_year}-{end_year}'.format(period, **meta[0])
```

Comment:
Would it make sense to move this logic in the parent function? There is something very similar there already.

Reply:
I wonder... then I have to pass it down through the chain. My intention here was to restrict the string creation for plots to the actual plot function. I leave it like it is for now. But that is just my preference if the good practice is to have less if statements and rather more function arguments that is absolutely okay with me